File: process_data.js
const fs = require('fs');

const files = [
    'data_kanto.js',
    'data_chubu.js',
    'data_kansai.js'
];

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');

    // 1. Remove onsen property
    content = content.replace(/,\s*onsen:\s*"[^"]*"/g, '');

    // 2. Remove gourmet property
    content = content.replace(/,\s*gourmet:\s*"[^"]*"/g, '');

    // 3. Format summary into bullet points
    content = content.replace(/summary:\s*"([^"]*)"/g, (match, text) => {
        let sentences = text.split('。').filter(s => s.trim().length > 0);
        let bullets = sentences.map(s => '・' + s.trim());
        let newHtml = bullets.join('<br>');
        return `summary: "${newHtml}"`;
    });

    fs.writeFileSync(file, content, 'utf8');
    console.log(`Processed ${file}`);
});

File: process_data.py
import re
import os

files = [
    'data_kanto.js', 
    'data_chubu.js', 
    'data_kansai.js'
]

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # 1. Remove onsen
    content = re.sub(r',\s*onsen:\s*"[^"]*"', '', content)
    
    # 2. Remove gourmet
    content = re.sub(r',\s*gourmet:\s*"[^"]*"', '', content)
    
    # 3. Format summary
    def replace_summary(match):
        text = match.group(1)
        sentences = [s.strip() for s in text.split('。') if s.strip()]
        if not sentences:
            if text.strip():
                return 'summary: "・' + text + '"'
            return 'summary: ""'
        bullets = ['・' + s for s in sentences]
        new_html = '<br>'.join(bullets)
        return 'summary: "' + new_html + '"'

    content = re.sub(r'summary:\s*"([^"]*)"', replace_summary, content)
    
    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)
        
    print(f'Processed {file}')
